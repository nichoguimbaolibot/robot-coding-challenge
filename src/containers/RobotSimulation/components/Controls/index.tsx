import React from 'react';
import Button from 'antd/lib/button';

interface IControllerProps {
  onChangeMovementToLeft: () => void;
  onChangeMovementToRight: () => void;
  onChangeRobotMovement: () => void;
  onReportRobotPosition: () => void;
}

const Controller: React.FunctionComponent<IControllerProps> = ({
  onChangeMovementToLeft,
  onChangeMovementToRight,
  onChangeRobotMovement,
  onReportRobotPosition,
}): JSX.Element => (
  <div className="direction-section">
    <h4 className="title">Movement and Positions</h4>
    <Button
      data-testid="button-move-left-test"
      className="left-button ui-button-primary"
      onClick={onChangeMovementToLeft}
    >
      LEFT
    </Button>

    <Button
      data-testid="button-move-right-test"
      className="right-button ui-button-primary"
      onClick={onChangeMovementToRight}
    >
      RIGHT
    </Button>

    <Button
      data-testid="button-move-test"
      className="move-button ui-button-alpha"
      onClick={onChangeRobotMovement}
    >
      MOVE
    </Button>

    <Button data-testid="button-report-test" onClick={onReportRobotPosition}>
      REPORT
    </Button>
  </div>
);

export default Controller;
