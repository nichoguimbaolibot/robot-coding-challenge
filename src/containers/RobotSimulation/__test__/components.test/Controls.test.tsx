import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import sinon from 'sinon';
import Controls from 'containers/RobotSimulation/components/Controls';

afterEach(cleanup);

describe('test Controls component', () => {
  const handleChangeMovementToLeft = sinon.spy();
  const handleChangeMovementToRight = sinon.spy();
  const handleChangeRobotMovement = sinon.spy();
  const handleReportRobotPosition = sinon.spy();

  it('renders CONTROLS component', () => {
    const { asFragment } = render(
      <Controls
        onChangeMovementToLeft={handleChangeMovementToLeft}
        onChangeMovementToRight={handleChangeMovementToRight}
        onChangeRobotMovement={handleChangeRobotMovement}
        onReportRobotPosition={handleReportRobotPosition}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('expect MOVE TO RIGHT event', () => {
    const testId = 'button-move-right-test';
    const { getByTestId } = render(
      <Controls
        onChangeMovementToLeft={handleChangeMovementToLeft}
        onChangeMovementToRight={handleChangeMovementToRight}
        onChangeRobotMovement={handleChangeRobotMovement}
        onReportRobotPosition={handleReportRobotPosition}
      />,
    );
    fireEvent.click(getByTestId(testId));
  });

  it('expect ROBOT MOVEMENT event', () => {
    const testId = 'button-move-test';
    const { getByTestId } = render(
      <Controls
        onChangeMovementToLeft={handleChangeMovementToLeft}
        onChangeMovementToRight={handleChangeMovementToRight}
        onChangeRobotMovement={handleChangeRobotMovement}
        onReportRobotPosition={handleReportRobotPosition}
      />,
    );
    fireEvent.click(getByTestId(testId));
  });

  it('expect MOVE TO LEFT event', () => {
    const testId = 'button-move-left-test';
    const { getByTestId } = render(
      <Controls
        onChangeMovementToLeft={handleChangeMovementToLeft}
        onChangeMovementToRight={handleChangeMovementToRight}
        onChangeRobotMovement={handleChangeRobotMovement}
        onReportRobotPosition={handleReportRobotPosition}
      />,
    );
    fireEvent.click(getByTestId(testId));
  });

  it('expect REPORT event', () => {
    const testId = 'button-report-test';
    const { getByTestId } = render(
      <Controls
        onChangeMovementToLeft={handleChangeMovementToLeft}
        onChangeMovementToRight={handleChangeMovementToRight}
        onChangeRobotMovement={handleChangeRobotMovement}
        onReportRobotPosition={handleReportRobotPosition}
      />,
    );
    fireEvent.click(getByTestId(testId));
  });
});
