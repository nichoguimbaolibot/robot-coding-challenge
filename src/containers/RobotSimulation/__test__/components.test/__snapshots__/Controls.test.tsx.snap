// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Controls component renders CONTROLS component 1`] = `
<DocumentFragment>
  <div
    class="direction-section"
  >
    <h4
      class="title"
    >
      Movement and Positions
    </h4>
    <button
      class="ant-btn left-button ui-button-primary"
      data-testid="button-move-left-test"
      type="button"
    >
      <span>
        LEFT
      </span>
    </button>
    <button
      class="ant-btn right-button ui-button-primary"
      data-testid="button-move-right-test"
      type="button"
    >
      <span>
        RIGHT
      </span>
    </button>
    <button
      class="ant-btn move-button ui-button-alpha"
      data-testid="button-move-test"
      type="button"
    >
      <span>
        MOVE
      </span>
    </button>
    <button
      class="ant-btn"
      data-testid="button-report-test"
      type="button"
    >
      <span>
        REPORT
      </span>
    </button>
  </div>
</DocumentFragment>
`;
